persistent/Database/Persist/TH.hs

```diff
@@ -1185,19 +1200,21 @@ fieldError tableName fieldName err = mconcat
     ]
 
 mkEntity :: EntityMap -> MkPersistSettings -> EntityDef -> Q [Dec]
-mkEntity entityMap mps entDef = do
-    entityDefExp <-
-        if mpsGeneric mps
-           then liftAndFixKeys entityMap entDef
-           else makePersistEntityDefExp mps entityMap entDef
-    let name = mkEntityDefName entDef
-    let clazz = ConT ''PersistEntity `AppT` genDataType
+mkEntity entityMap mps preEntDef = do
+    entityDefExp <- liftAndFixKeys entityMap preEntDef
+    let
+        entDef = fixEntityDef preEntDef
+        genDataType = genericDataType mps entName backendT
+        entName = entityHaskell entDef
+        name = mkEntityDefName entDef
+        clazz = ConT ''PersistEntity `AppT` genDataType
+
     tpf <- mkToPersistFields mps entDef
     fpv <- mkFromPersistValues mps entDef
     utv <- mkUniqueToValues $ entityUniques entDef
     puk <- mkUniqueKeys entDef
     let primaryField = entityId entDef
-    fields <- mapM (mkField mps entDef) $ primaryField : getEntityFields entDef
+    fields <- mapM (mkField mps entDef) $ primaryField : getEntityFieldsDatabase entDef
```

Comment:
Since migrations are now handled by `mkEntityDefList`, we need to preserve *all* the field information.